bcgov/cas-registration
# feat: View and manage user access requests as an Op/Org admin

Addresses [#28](https://app.zenhub.com/workspaces/team-moose-sprint-board-60d216175a2689000ec774d0/issues/gh/bcgov/cas-registration/28). Adds backend API and frontend interface for an Organization Admin to view, approve/deny user access requests.

## Changes 🚧

- Add GET endpoint to retrieve all user access request statuses for a particular Operator `/operators/{operator_id}/user-operators`.
- Add GET endpoint to retrieve all of a user's associated Operators `/get-users-operators/{user_id}`.
- Add PUT endpoint to change access request status of a UserOperator `/select-operator/user-operator/{user_id}/update-status`.
- Add page for `/dashboard/users` for External Admin.
  - Lists all users for External Admin's Operators and their access request statuses.
  - Ability to Approve/Deny/Undo (set to pending) user acceses requests.
 
## To test 🔬

Spin up the front and backend
Login using `Business BCeID development environment` account from 1Pass in your local environment 
Navigate to `/dashboard/users`. You should see a list of users with various statuses. 
- The user with the _Myself_ status should have no actions available
- The user with the _Pending_ status should have the Approve and Deny actions
- A user that is _Approved_ or _Denied_ should have the Undo action

## Notes📝 
Out of scope of this ticket was the ability to change the User's Role, but this functionality is part of the Wireframes.
